core/src/main/java/com/graphhopper/routing/util/parsers/BikeCommonAverageSpeedParser.java

```diff
@@ -119,6 +119,8 @@ protected BikeCommonAverageSpeedParser(DecimalEncodedValue speedEnc, EnumEncoded
      */
     double applyMaxSpeed(ReaderWay way, double speed, boolean bwd) {
         double maxSpeed = getMaxSpeed(way, bwd);
+        if (way.getTag("highway", "").equals("living_street"))
+            return !isValidSpeed(maxSpeed) ? highwaySpeeds.get("living_street") : maxSpeed <= highwaySpeeds.get("cycleway") ? maxSpeed : highwaySpeeds.get("cycleway");
```

Comment:
> We should either avoid this or make it consistent for other highway types like service.

I checked other highway types for which we assume a smaller speed compared to the 18km/h for `cycleway`. You are right. For `service`, `path`, `track`, `pedestrian`, and `footway`  the combination with presence of the `maxspeed` tag seems to make sense in most cases. I checked this with overpass. For the few `steps` these must be tagging errors, but they are so few that these cases can be ignored.

> I'm in favour of removing this to keep it simple and rely on the highway type and tracktype/surface alone to define the speed and use maxspeed only to decrease it.

Now that I saw in overpass how many cases these are I prefer the little more complex, but I believe better and generic solution. This solution should work that same way as I tried for the `living_street`, but irrespective of the highway type, and also taking the limitation of the tracktype/surface/smoothness into account so that these limits never get overwritten. Or do you think that this would become too complex?

> Handling higher defaults speeds for living_street should be done elsewhere.

Could you please specify what you mean by "elsewhere", I don't understand where this could be.

Reply:
Please see [52f4496](https://github.com/graphhopper/graphhopper/pull/3032/commits/52f4496c3008de06d6e46c91fc59ab51a370a854). If you agree with this I will change the title of this PR into "Run bicycles at tagged maxspeed if allowed and possible"

Reply:
> Do you know what I mean?

Sorry, but I cannot quite follow your concerns. The modifications change the behavior only for ways for which we assume a lower highway speed compared to the tagged `maxspeed` . So the change is effective only for `living_street`, `service`, `path`, `track`, `pedestrian` , `road` and `footway`. `steps` are excluded in the code. 

>Because afterwards roads with a max_speed tag will be very likely preferred which is not what we want.

As you choose the term `roads` in plural here I believe that you thought the other more car centric highway types might also be covered. I don't think that this is the case. Or did you mean the `road` highway type only?

Would it be possible to name a situation by which you can illustrate your concerns more specifically? Maybe by an additional test?

> The main criteria for bike routing should stay the highway and surface/tracktype tags.

Agreed, but I think that this is still the case. And we are already considering the `maxspeed` as an upper limit.

Reply:
@karussell May I ask you to look at the last questions, it looks as you missed it. Thanks in advance!

Reply:
>> Couldn't we increase the speed for living_street to 6km/h and keep the rest unchanged to improve the general unnecessary avoidance of living_street due to slow speed?

OK, even tough I consider this more a hack I changed the code according to this proposal. Please check again.